### PR TITLE
fix(utils): preserve non-latin characters in slug generation

### LINKED ISSUE #1036: Handle non-Latin characters in slug generation
## Problem

`toSlug()` uses the `slug` npm package which strips non-Latin characters (Japanese, Chinese, Korean, etc.). This causes unique constraint violations when multiple titles differ only in non-Latin characters.

For example, these chapter titles all produce the same slug (`estruturas-com`):

- "Estruturas com わけ・はず・もの"
- "Estruturas com よう・みたい・らしい"
- "Estruturas com ところ・まま・ばかり"
- "Estruturas com しか・さえ・こそ"

This broke the `addChaptersStep` workflow with `Unique constraint failed on the fields: (course_id, slug)`.

## Proposed Solution

Replace the `slug` package with [`slugify`](https://www.npmjs.com/package/slugify), which preserves Unicode characters by default:

```typescript
import slugify from "slugify";

export function toSlug(str: string): string {
  return slugify(str.trim(), { lower: true, strict: true });
}
```

This produces unique, readable slugs that preserve the original characters:

- "Estruturas com わけ・はず・もの" → `estruturas-com-わけはずもの`
- "Estruturas com よう・みたい・らしい" → `estruturas-com-ようみたいらしい`
- "Condicional com と" → `condicional-com-と`
- "Condicional com たら" → `condicional-com-たら`

### Why Unicode slugs over transliteration?

- **Transliteration is lossy and unreliable for CJK** — kanji have multiple readings (日 could be "nichi", "hi", "jitsu", "ka"), so transliterated slugs may not match what a native speaker expects
- **Simpler** — no extra transformation step or transliteration dependency
- **Better for a learning platform** — users are studying these characters, so seeing them in URLs is informative
- **Modern standard** — Wikipedia, Django, Rails all support Unicode slugs; browsers and Next.js handle them natively

Since `toSlug` is used across 33 files, fixing it at the source handles all current and future slug generation automatically.

## Acceptance Criteria

- [ ] Replace `slug` with `slugify` in `packages/utils`
- [ ] `toSlug` preserves non-Latin characters (Japanese, Chinese, Korean, etc.)
- [ ] Existing tests updated to cover non-Latin input
- [ ] No unique constraint violations when titles differ only in non-Latin characters

### PR DESCRIPTION
**Pull request by ceolinwill**
## Summary

- Replace `slug` npm package with custom Unicode-aware `toSlug` using `\p{L}`, `\p{N}`, `\p{M}` regex classes — preserves CJK, Cyrillic, Arabic, Thai, and other scripts in slugs
- Add `deduplicateSlugs` utility with collision-safe suffix generation, applied in `addChaptersStep` and `addLessonsStep` batch creates
- Enforce `SLUG_MAX_LENGTH` inside `toSlug` so all callers get truncation automatically (removed redundant manual truncation from `step-image.ts` and `step-visual-image.ts`)
- Remove `slug` and `@types/slug` dependencies

Closes #1036

## Test plan

- [x] 271 unit tests pass (comprehensive slug tests for Latin accents, CJK, mixed scripts, punctuation, edge cases, deduplication with collision avoidance)
- [x] All apps build (main, editor, api)
- [x] E2E tests pass for all apps
- [x] `pnpm typecheck`, `pnpm knip --production`, `pnpm turbo quality:fix` all clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Replaced the `slug` package with a Unicode‑aware `toSlug` that preserves non‑Latin characters, and added `deduplicateSlugs` to prevent collisions in batch chapter/lesson creates. Addresses #1036.

- **Bug Fixes**
  - Preserve CJK, Cyrillic, Arabic, Thai, and mixed scripts; remove punctuation; collapse dashes; normalize accents; enforce a 50‑char `SLUG_MAX_LENGTH`.
  - Strip trailing hyphens after truncation.
  - Apply `deduplicateSlugs` in `addChaptersStep` and `addLessonsStep` to suffix duplicates (`-1`, `-2`, …).
  - Remove manual truncation from step image generators since `toSlug` now truncates.

- **Dependencies**
  - Remove `slug` and `@types/slug` from `packages/utils`.

<sup>Written for commit f6525d3aa8ee6a7825c8b899bfa1073bfcdee2ca. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/utils/src/string.ts [hunk 1]
```diff
@@ -12,7 +10,40 @@ export function normalizeString(str: string): string {
...
+}
+
+function nextAvailableSlug(base: string, taken: Set<string>): string {
+  let counter = 1;
+
+  while (taken.has(`${base}-${counter}`)) {
+    counter += 1;
+  }
+
+  return `${base}-${counter}`;
```
[the remark this answers is not in the document]
Comment (ceolinwill) [reply packages/utils/src/string.ts:30]: SLUG_MAX_LENGTH is a soft constraint to avoid long slugs. it's fine to have suffixes
